WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java

```diff
@@ -207,6 +208,10 @@ public static ReaderPost fromJson(JSONObject json) {
             post.setRailcarJson(jsonRailcar.toString());
         }
 
+        // set the card type last since it depends on information contained in the post - note
+        // that this is stored in the post table rather than calculated on-the-fly
```

Comment:
Yes, posts are parsed in the background - see [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/services/ReaderPostService.java#L278), for example.